### PR TITLE
Better sync outputs

### PR DESCRIPTION
**Pull request by OpenBagTwo**
## Summary

Addresses #57 by:
- summarizing rsync reports in a way that's useful and customizable
- not suppressing stdout when doing an actual sync and instead making use of some fancy rsync flags to make the output concise and useful (with progress bars!)

## List of Changes

* Refactors the non-impl-specific sync utilities into `sync.utils` instead of leaving them in `__init__.py`
* Leverages a bunch of fancy flags in `rsync` to customize the output--rather than suppressing it--during syncs
* Leverages _separate_ fancy flags in `rsync` to allow EnderChest to _summarize_ the reports of dry runs
* Allows verbosity to be accessed by CLI "actions" that want it
* Adds a new intermediate log level in between INFO and WARNING so that `-q` doesn't have to mean "completely silent"

## To Do
- [x] I need to make sure that all the options used are available on the rsync that ships with all my target platforms. That's part of the reason I'm opening this PR now (test the MacOS runner)
- [x] Document the outputs at the different log levels
- [x] Verbosity tests!

## Validation Performed

* Been informally dogfooding this a bit as I customize the output messages, but I definitely need to be thorough and provide examples of each of the log levels

## PR Type

- [ ] This PR introduces a breaking change (will
  [require a bump in the minor version](https://semver.org/))
- [ ] The changes in this PR are high urgency and necessitate a hotfix or patch
  release (will require rebasing off of `release`)
- [ ] This is a release (staging) PR (maintainer use only)

## Checklist:

- [x] I have read [the contributor's guide](https://openbagtwo.github.io/EnderChest/dev/contrib/)
- [x] I have run `mkdocs serve` locally and ensured that all API docs and
  changes I have made to the static pages are rendering correctly, with all links
  working
- [ ] All tech debt concerns have been resolved, documented as issues, or otherwise
  accepted
- [x] I agree to license my contribution to this project under
  [the GNU Public License v3](https://www.gnu.org/licenses/gpl-3.0.en.html)

### CODE REVIEW

**Comment by OpenBagTwo** [comment]
Documentation on the verbosity levels is covered generically in the CLI docs:

https://github.com/OpenBagTwo/EnderChest/blob/d412eeea57a39462df94c4dab1069a3550d3984f/enderchest/cli.py#L244-L257

and explicitly in the rsync API docs:

https://github.com/OpenBagTwo/EnderChest/blob/d412eeea57a39462df94c4dab1069a3550d3984f/enderchest/sync/rsync.py#L54-L68

(and they look great rendered via `mkdocs serve`)

If `--info=progress2` isn't implemented on the latest macOS runner, I'll know in a minute, but I noticed it was accepted on my (desperately-in-need-of-upgrading) Raspberry Pi which is still running 32-bit Raspbian (from before [they changed the name to Raspberry Pi OS](https://www.tomshardware.com/news/raspberry-pi-os-no-longer-raspbian)), which, btw, doesn't even recognize `rsync -V` (using `rsync --version` spat out `rsync  version 3.1.3`), so :crossed_fingers:.

I will probably go ahead and check my own ARM and Intel macs before merging this PR. Windows is out of scope (#67) and has bigger problems right now anyway (https://github.com/OpenBagTwo/EnderChest/issues/67#issuecomment-1580664276).

**Comment by OpenBagTwo** [comment]
Haha, `rsync  version 2.6.9  protocol version 2` on my 2013 MBP, which still runs Mojave, didn't even support `-shaz`. However:

1. I could install a modern rsync using mamba
1. I could install a modern rsync using `brew`

even though brew is beyond fed up with my refusal to upgrade:

> Warning: You are using macOS 10.14.
> We (and Apple) do not provide support for this old version.
> It is expected behaviour that some formulae will fail to build in this old version.

(the reason I'm sticking with Mojave is that [Catalina dropped support for 32-bit](https://www.theverge.com/2019/10/12/20908567/apple-macos-catalina-breaking-apps-32-bit-support-how-to-prepare-avoid-update), and since this ain't my daily driver, I'm _beyond fine_ holding my machine back)

**Bottom line:** I'm going to amend the docs to explicitly state that [rsync 3.2 or greater is required](https://download.samba.org/pub/rsync/NEWS#3.2.0) to use rsync with EnderChest (even though 3.1 would probably work)

**Comment by OpenBagTwo** [comment]
Just checked my 2020 Macbook Air (running Monterrey) and... it's also running rsync 2.6.9...

[Apple, this is just pathetic](https://dev.to/al5ina5/updating-rsync-on-macos-so-you-re-not-stuck-with-14-year-old-software-1b5i).

So beyond just stating that rsync 3.2 is required, I should probably explicitly include instructions for upgrading it, or at least the above link...
